### ISSUE TITLE
Faster Column count()

### ISSUE THREAD

**Issue by vsraptor** [ilanschnell/bitarray#162]
I'm implementing 2D array in Cython

```
shape: (10000,10000)
mx.count(1)       '.sum()': {'sa': '5.68 ms'},
mx.count(1, ifrom, ito)          '.sum(axis=0)': {'sa': '9.39 ms'},
mx[start:: row_size].count(1)         '.sum(axis=1)': {'sa': '907 ms'},

```
is it possible to have faster-step  .....  .count(from,to,**STEP**)  ... so I can lower this 907ms..
The axis=0 case sped up with 3ms when i switched to .count()

-----

One more proposal for .util method to copy fast,  parts of one 2D ary to another array:

```
		for i,r in enumerate(row_iter) :
			for j,c in enumerate(col_iter) :
				ix = r * self.ncols + c
				nix = i * new.ncols + j
				new.mx[nix] = self.mx[ix]
```

this works for slice converted to range, list, numpy array, range ..

`loop2D(src, dst, row_iter, col_iter)`

or if this wont be fast enough, then something that can be converted to simple for-loops

`loop2D(src, dst, ((s1,e1,step1), (s2,e2,step2)) )`

this will may be need : assert (e1-s1)/step1 > (e2-s2)/step2

-----

 '[100:120,100:120]': {'sa': '39.9 µs'},
 '[5,5]': {'sa': '164 ns'},
 '[5,5]=1': {'sa': '118 ns'},
 '[:,[9737, 2242, 8392, 1553, 9085, 2709, 5124, 5190, 5098, 2336]],': {'sa': '7.94 ms'},
 '[:100,:100]': {'sa': '613 µs'},
 '[[0,1,2],[0,1,2]]': {'sa': '1.12 µs'},
 '[[8421, 1376, 2939, 1110, 972, 1711, 4016, 2225, 4497, 7693],:],': {'sa': '7.2 ms'},
 '[[8421, 1376, 2939, 1110, 972, 1711, 4016, 2225, 4497, 7693],:].sum(axis=1),': {'sa': '482 ''ms'},
 '[[8421, 1376, 2939, 1110, 972, 1711, 4016, 2225, 4497, 7693],[9737, 2242, 8392, 1553, 9085, 2709, 5124, 5190, 5098, 2336]],': {'sa': '4.67 ' 'µs'}}

**Comment by ilanschnell** [comment]
Thank you for using bitarray and these proposals!  An optional `step` argument to the `.count()` method should not be hard implement.  In the meantime, I suggest using slicing:
```
a[start:stop:step].count()
```
Even though this operation creates an intermediate bitarray, it will still be much faster than counting the values in a Python loop.

Your second proposal is not entirely clear to me.  Wouldn't you be able to do this using slicing?

**Comment by vsraptor** [comment]
loop2D(src, dst, ((s1,e1,step1), (s2,e2,step2)) )

the idea is to treat the array's as 2D and copy the content of src array slice to dst slice.

I agree it is too specific. ;)

**Comment by ilanschnell** [comment]
Thanks for the clarification.  Yes this is very specific.  You could write your own C extensions to do things like this.  The way you would do that is basically the same way the `bitarray.util` extension is implemented.

BTW: I just released bitarray 2.3.7 which includes the new step argument for the `.count()` method.

**Comment by vsraptor** [comment]
how do i skip end-of-array argument ...

ary.count(1, start=c, step=nrows)
